### PR TITLE
feature: add migration tool for reviving application contract

### PR DESCRIPTION
**Pull request by Vixtir**

### CODE REVIEW

File: src/main/scala/io/hydrosphere/serving/manager/domain/application/migrations/ApplicationSignatureMigrationTool.scala [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+              depNames = graph.stages.flatMap(_.variants).toList.flatMap(_.requiredDeployConfig)
+              deployments <-
+                depNames
+                  .traverse(deploymentRepo.get)
+                  .map(list =>
+                    list.collect {
+                      case Some(dc) => dc.name -> dc
+                    }.toMap
+                  )
+              _ <- restoreApp(appRow, versions, servables, deployments) match {
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/domain/application/migrations/ApplicationSignatureMigrationTool.scala:77]: ❌  `restoreApp` is an effectful function, right? So, if I put `restoreApp[F](...)` how is it possible to pattern match against `Left` and `Right`?

**Review by KineticCookie — CHANGES_REQUESTED** [review]

File: src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBApplicationRepository.scala [hunk 2]
```diff
@@ -342,32 +342,41 @@ object DBApplicationRepository {
...
           servableMap = servables.map(x => x.fullName -> x).toMap
           deploymentNames <- F.fromEither(
-            nodesOrError.map(n =>
-              n.map(y =>
-                y.requiredDeployConfig match {
-                  case Some(dc) => dc
-                }
-              )
-            )
+            nodesOrError.map(graphServable => graphServable)
```
[the remark this answers is not in the document]
Comment (Vixtir) [reply src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBApplicationRepository.scala]: fixed

File: src/main/scala/io/hydrosphere/serving/manager/domain/application/migrations/ApplicationSignatureMigrationTool.scala (hunk 1 above)
Comment (Vixtir) [reply src/main/scala/io/hydrosphere/serving/manager/domain/application/migrations/ApplicationSignatureMigrationTool.scala:76]: Made it as function which returns Either[DomainError, Application]

**Review by KineticCookie — APPROVED** [review]
LGTM 👍

**Pull request merged**